plugin/src/main/java/org/opensearch/ml/action/stats/MLStatsNodesTransportAction.java

```diff
@@ -126,22 +155,80 @@ MLStatsNodeResponse createMLStatsNodeResponse(MLStatsNodesRequest mlStatsNodesRe
+            searchHiddenModels(ActionListener.wrap(hiddenModels -> {
+                for (String modelId : mlStats.getAllModels()) {
+                    if (isSuperAdmin || !hiddenModels.contains(modelId)) {
+                        if (mlStatsInput.retrieveStatsForModel(modelId)) {
+                            Map<ActionName, MLActionStats> actionStatsMap = new HashMap<>();
+                            for (Map.Entry<ActionName, MLActionStats> entry : mlStats.getModelStats(modelId).entrySet()) {
+                                if (mlStatsInput.retrieveStatsForAction(entry.getKey())) {
+                                    actionStatsMap.put(entry.getKey(), entry.getValue());
+                                }
+                            }
+                            modelStats.put(modelId, new MLModelStats(actionStatsMap));
                         }
                     }
-                    modelStats.put(modelId, new MLModelStats(actionStatsMap));
                 }
-            }
+            }, e -> { throw new OpenSearchStatusException("Model search wasn't successful", RestStatus.INTERNAL_SERVER_ERROR); }));
         }
-
         return new MLStatsNodeResponse(clusterService.localNode(), statValues, algorithmStats, modelStats);
     }
+
+    @VisibleForTesting
+    void searchHiddenModels(ActionListener<Set<String>> listener) {
+        SearchRequest searchRequest = buildSearchRequest();
+        // Use a try-with-resources block to ensure resources are properly released
+        try (ThreadContext.StoredContext threadContext = client.threadPool().getThreadContext().stashContext()) {
+            // Wrap the listener to restore thread context before calling it
+            ActionListener<Set<String>> internalListener = ActionListener.runBefore(listener, () -> threadContext.restore());
+            // Wrap the search response handler to handle success and failure cases
+            ActionListener<SearchResponse> al = ActionListener.wrap(response -> {
+                // Initialize the result set
+                Set<String> result = new HashSet<>(response.getHits().getHits().length); // Set initial capacity to the number of hits
+
+                // Iterate over the search hits and add their IDs to the result set
+                for (SearchHit hit : response.getHits()) {
+                    result.add(hit.getId());
+                }
+                // Notify the listener of the search results
+                internalListener.onResponse(result);
+            }, e -> {
+                // Notify the listener of any search failures
+                internalListener.onFailure(e);
+            });
+
+            // Execute the search request asynchronously
+            client.search(searchRequest, al);
+        } catch (Exception e) {
+            // Notify the listener of any unexpected errors
+            listener.onFailure(e);
+        }
+    }
+
+    private SearchRequest buildSearchRequest() {
```

Comment:
Can we improve the function name so its clear this is building a search request for finding hidden model, so it can be like `buildHiddenModelSearchRequest`